jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -5307,4 +5307,29 @@ public IntStream codePoints() {
 		return CharSequence.super.codePoints();
 	}
 /*[ENDIF]*/	
+
+/*[IF Java10]*/
+	/*
+	 * Internal API, assuming no modification to the byte array returned.
+	 * In case of enableSharingInSubstringWhenOffsetIsZero is set to true (by default), the actual bytes will be 
+	 * copied to a new array, otherwise original byte array value is returned.
+	 */
+	byte[] value() {
+		int	validLen;
+		if (enableCompression) {
+			if (LATIN1 == coder) {
```

Comment:
we should avoid testing coder here - we want to test:
`if (null == compressionFlag || count >= 0) {`
because the JIT can nop this if until a decompressed string is created when compression is enabled while we can't nop the coder field test.

Reply:
the special noping is specifically for compressionFlag - this is an optimization where we happen to know that the field is of a class that will not be loaded until the flag is set and so we use class loading as the proxy - we do not do this transform in general - it is an opt specific to improving the performance when we are dealing with uniformly represented strings (all compressed or all decompressed)